src/web/js/dashboard/StudentList.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react';
+import Student from './Student.js';
+import { Button, Textfield, Card, CardTitle, CardText, Spinner } from 'react-mdl';
+
+class StudentList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      addingStudent: false,
+      newStudentFirstName: '',
+      newStudentLastName: '',
+      newStudentEmail: ''
+    };
+  }
+
+  handleClickAddStudent = () => {
+    this.setState({addingStudent: ! this.state.addingStudent});
+  }
+
+  handleChange = (event) => {
+    this.setState({[event.target.id]: event.target.value});
+  }
+
+  handleSubmitAddStudent = (event) => {
+    event.preventDefault();
+    this.props.snackBar('Student Added. Please allow a few seconds for changes to appear.');
+    this.setState({addingStudent: false});
+    this.props.api.addStudent({
+      firstName: this.state.newStudentFirstName.trim(),
+      lastName: this.state.newStudentLastName.trim(),
+      email: this.state.newStudentEmail.trim()
+    }).then((resp) => {
```

Comment:
What happens if this promise chain fails?  Is it just invisible failure to the user?